tox.ini

```diff
@@ -86,3 +86,10 @@ skip_install = true
 commands =
     black --config=.black {toxinidir}
     flake8
+
+[testenv:clean]
+description = Remove all that tox stuff
+allowlist_externals = rm
+skip_install = true
+commands =
+  rm -rf .tox/ ibcdfo_pypkg/.tox/ ibcdfo_pypkg/build/ ibcdfo_pypkg/.coverage_ibcdfo ibcdfo_pypkg/benchmark_results
```

Comment:
I don't like the idea of using relative paths since I do not understand how `tox` deals with the current working directory.  That said, you are explicitly deleting folders that are likely of no importance to developers.  Therefore, if the CWD ends up being something other than what you intended, it either won't find the folder to delete or did little damage.

Aside from a rogue developer slipping something nefarious into this line, which is always a risk, is that a fair risk assessment?

If `ibcdfo_pypkg/benchmark_results` is being produced by one of your tests, then I think that it would be better for the test to remove it or for the test class to use `tearDown` to remove it.  The only issue with that is if a test fails and you want to use a result to debug, you'd have to comment out `tearDown` to access the results.

I think that we might have discussed this, but in the future I think that the `tox.ini` of each python package should have its own `clean` and this master `tox.ini` should somehow invoke these.  This gives more control and flexibility to the developers of each package.

I like that my comments are a more substantial percentage of the content of the PR than the changes themselves.

Reply:
I agree that relative paths aren't preferred, but I'm not sure about the alternative (unless there is some `tox_root` variable that I'm unaware of. Otherwise, yes, I think your risk assessment is correct.

As for the `benchmark_results` directory, I wouldn't want the test to remove them automatically. Since this isn't "`tox` stuff" necessarily, I will remove it from the `clean` statement. 

I agree that when there are multiple packages in a repo, each tox could have a different `clean` statement for greater flexibility. When there is only one package (as is the case in IBCDFO currently) then I think it's easier to have one `clean` to deal with.

Reply:
Too little, too late: https://tox.wiki/en/latest/config.html#toxinidir